src/amazonqFeatureDev/errors.ts

```diff
@@ -64,6 +70,18 @@ export class ContentLengthError extends ToolkitError {
     }
 }
 
+export class UnknownApiError extends ToolkitError {
+    constructor(message: string, api: string) {
+        super(message, { code: `${api}-Unknown` })
+    }
+}
+
+export class ApiError extends ToolkitError {
+    constructor(message: string, api: string, errorName: string, errorCode: number) {
+        super(message, { code: `${api}-${errorName}-${errorCode}` })
```

Comment:
off-topic/future reference: we need to enhance the default handling so that http status codes are automatically captured from the thrown exception and stored in the metric's  https://github.com/aws/aws-toolkit-vscode/blob/af655f604ffc8fa99a49a901636dd270175b5c2e/src/shared/telemetry/spans.ts#L151-L161

Ref: https://sim.amazon.com/issues/IDE-11056